.github/workflows/industrial_ci_action.yml

```
@@ -0,0 +1,36 @@
+# This config uses industrial_ci (https://github.com/ros-industrial/industrial_ci.git).
+# For troubleshooting, see readme (https://github.com/ros-industrial/industrial_ci/blob/master/README.rst)
+
+name: BuildAndTest
+
+# This determines when this workflow is run
+on:
+  # When ros2 branch is pushed to
+  push:
+    branches: [ ros2 ]
+  # When there is a pull request against ros2
+  pull_request:
+    branches: [ ros2 ]
+
+jobs:
+  industrial_ci:
+    strategy:
+      matrix:
+        env:
+          - {ROS_DISTRO: foxy, ROS_REPO: main}
+          - {ROS_DISTRO: foxy, ROS_REPO: testing}
+          - {ROS_DISTRO: foxy, ROS_REPO: main, CLANG_TIDY: pedantic}
```

What about combining these?
```suggestion
          - {ROS_DISTRO: foxy, ROS_REPO: testing, CLANG_TIDY: pedantic}
```